tool/tsh/kube_proxy.go

```diff
@@ -194,43 +204,66 @@ func makeKubeLocalProxy(cf *CLIConf, tc *client.TeleportClient, clusters kubecon
 		return nil, trace.Wrap(err)
 	}
 
-	cas, err := loadKubeLocalCAs(profile, clusters.TeleportClusters())
+	keyPem, err := utils.ReadPath(profile.KeyPath())
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	localClientKey, err := keys.ParsePrivateKey(keyPem)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	cas, err := createKubeLocalCAs(localClientKey, clusters.TeleportClusters())
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 	lpListener, err := alpnproxy.NewKubeListener(cas)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+
+	kubeProxy := &kubeLocalProxy{
+		tc:        tc,
+		profile:   profile,
+		clusters:  clusters,
+		clientKey: localClientKey,
+		localCAs:  cas,
+	}
+
 	localProxy, err := alpnproxy.NewLocalProxy(
 		makeBasicLocalProxyConfig(cf, tc, lpListener),
-		alpnproxy.WithHTTPMiddleware(alpnproxy.NewKubeMiddleware(certs)),
+		alpnproxy.WithHTTPMiddleware(alpnproxy.NewKubeMiddleware(certs, kubeProxy.getCertReissuer(tc), nil)),
 		alpnproxy.WithSNI(client.GetKubeTLSServerName(tc.WebProxyHost())),
 		alpnproxy.WithClusterCAs(cf.Context, tc.RootClusterCACertPool),
 	)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+	kubeProxy.localProxy = localProxy
 
 	forwardProxy, err := alpnproxy.NewKubeForwardProxy(cf.Context, port, localProxy.GetAddr())
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+	kubeProxy.forwardProxy = forwardProxy
 
 	kubeConfigPath := os.Getenv(proxyKubeConfigEnvVar)
 	if kubeConfigPath == "" {
 		_, port, _ := net.SplitHostPort(forwardProxy.GetAddr())
 		kubeConfigPath = path.Join(profile.KubeConfigPath(fmt.Sprintf("localproxy-%v", port)))
 	}
+	kubeProxy.kubeConfigPath = kubeConfigPath
```

Comment:
nit:
```suggestion
	kubeProxy.kubeConfigPath = os.Getenv(proxyKubeConfigEnvVar)
	if kubeConfigPath == "" {
		_, port, _ := net.SplitHostPort(forwardProxy.GetAddr())
		kubeProxy.kubeConfigPath = path.Join(profile.KubeConfigPath(fmt.Sprintf("localproxy-%v", port)))
	}
```

same for kubeProxy.forwardProxy